Chris add GitHub rules

chrispsommers opened the pull request:
Signed-off-by: Chris Sommers <chris.sommers@keysight.com>

Adapted from https://github.com/Azure/DASH/blob/main/doc-github-rules.md which I co-authored with MSFT.
Per Slack thread: https://opi-project.slack.com/archives/C033E418VCK/p1649275592758869. Other opi projects can refer to this.

mestery commented on doc-github-rules.md:
```diff
@@ -0,0 +1,106 @@
+
+If you use the Github desktop app, see [Creating an issue or pull request](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/working-with-your-remote-repository-on-github-or-github-enterprise/creating-an-issue-or-pull-request). 
+
+## Create a Pull Request from a branch in the original repository
```
I would prefer we remove this as I would rather we only do PRs from forks and not from the original repository.

chrispsommers replied:
Done. I agree, that had been a concession to the original project, which uses local branches more than most Community projects.

mestery commented on doc-github-rules.md:
```diff
@@ -0,0 +1,106 @@
+Pull requests generally are open to community review and get accepted as-is or with requested changes. The more impactful the changes, the more review activity is to be expected.
+
+## PR Accept Phase
+Once a PR has received sufficient review and concerns are satisfied, a designated person with sufficient [permissions](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) can accept the Pull Request. 
```
Maybe link to the [maintainers](https://github.com/opiproject/opi/blob/main/MAINTAINERS.md) file here?

chrispsommers replied:
Done, thanks for suggestion.

mestery commented on doc-github-rules.md:
```diff
@@ -0,0 +1,106 @@
+Once a PR has received sufficient review and concerns are satisfied, a designated person with sufficient [permissions](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) can accept the Pull Request. 
+
+## Directly Editing Artifacts
+Contributors with write access permission or higher have rights to directly edit documents and push changes. This practice should be used very sparingly, and only in the draft stages of documents, or to make trivial changes in documents already in use.
```
Excellent wording, I agree with this!

mestery reviewed (changes requested):
Good start, some suggestions included in my review.

chrispsommers commented:
@mestery I addressed all your concerns and resolved all the conversations. Yet for some reason, the PR shows requested changes still need to be addressed.

mestery reviewed (approved):
Excellent work @chrispsommers, thank you!

glimchb reviewed (approved):
lgtm
